2009-05-05  Aurelien Jarno  <aurelien@aurel32.net>

	* resolv/res_query.c (__libc_res_nquery): If one query returns NOTIMP
	or FORMERR and the other NOERROR, don't raise an error.

--- a/resolv/res_query.c
+++ b/resolv/res_query.c
@@ -289,6 +289,10 @@ __libc_res_nquery(res_state statp,
 			break;
 		case FORMERR:
 		case NOTIMP:
+			if ((hp->rcode == NOERROR && ntohs (hp->ancount) != 0)
+			    || (hp2->rcode == NOERROR
+				&& ntohs (hp2->ancount) != 0))
+				goto success;
 		case REFUSED:
 		default:
 			RES_SET_H_ERRNO(statp, NO_RECOVERY);
